osm_cli/converters/pdf_converter.py

```diff
@@ -0,0 +1,62 @@
+import socket
+
+import docker
+from docker.errors import DockerException
+from pydantic import FilePath
+import requests
+
+from osm_cli.converters.converter import Converter
+from osm_cli.utils.config import config
+
+
+class PDFConverter(Converter):
+    protocol: str = config.PROTOCAL
+    host: str = config.HOST
+    port: int = config.PORT
+
+    def convert(self, pdf_path: FilePath) -> str:
+        """Convert a PDF file to XML using ScienceBeam Parser.
+
+        Args:
+            pdf_path: Path to the PDF file
+        Returns:
+            XML content as a string
+        """
+        sciencebeam_url: str = f'{self.protocol}://{self.host}:{self.port}/api/convert'
+        with open(pdf_path, 'rb') as pdf_file:
+            files = {'file': pdf_file}
+            headers = {'Accept': 'application/tei+xml'}
+            response = requests.post(
+                sciencebeam_url, files=files, headers=headers)
+
+            if response.status_code == 200:
+                return response.text
+            else:
+                response.raise_for_status()
+
+    def is_host_ready(self, timeout=3) -> bool:
+        """Check if the host is ready to accept requests.
+        Args:
+            timeout: Timeout in seconds
+        Returns:
+            True if the docker is host is ready, False otherwise
+        """
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            sock.settimeout(timeout)
+            try:
+                sock.connect((self.host, self.port))
+            except (socket.timeout, socket.error):
+                return False
+            return True
+
+    def is_docker_running(self):
```

Comment:
This function might be misnamed. To me it checks if the sciencebeam parser image is available locally. I guess it implicitly confirms that there is a docker daemon running but perhaps we also want to check that there is a sciencebeam container running? The is_host_ready method would check that already though right?

Reply:
Got it. We removed the needless method `is_docker_running`.